### DISCUSSION TITLE
Fail to open Jupyter Notebook

### DISCUSSION THREAD

**Discussion by horlust** [jupyter/notebook#7448] [Q&A]
Hi everyone,

I am using Ubuntu 22.04.3 LTS on Windows. Whenever I try to open Jupyter Notebook from the Terminal, I get: _No web browser found: Error('could not locate runnable browser')._ When I click on the link, it asks for a password or token. I've tried definying a password with _jupyter notebook password_, but when I put the defined password I get _invalid credentials_. What am I doing wrong?

**Comment by yadavnikhil03** [comment]
### **Step 1: Manually Open Jupyter Notebook**
1. Open your WSL terminal.
2. Run:
`jupyter notebook --no-browser`
3. You will see an output with a URL like this:
`http://127.0.0.1:8888/?token=your_long_token_here`
Copy that URL and paste it into your Windows browser.

### **Step 2: Avoid Token Login (Use a Password)**
_If you want to set a password instead of using the token_
1. Run 
`jupyter notebook password`
2. Enter a new password.
3. Restart Jupyter Notebook
`jupyter notebook --no-browser`

**If this solution worked for you, please mark it as accepted!** 😊
